core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -749,9 +749,11 @@ class ReplicaManager(val config: KafkaConfig,
    * @param responseCallback              callback for sending the response
    * @param delayedProduceLock            lock for the delayed actions
    * @param recordValidationStatsCallback callback for updating stats on record conversions
-   * @param requestLocal                  container for the stateful instances scoped to this request
-   * @param transactionalId               transactional ID if the request is from a producer and the producer is transactional
+   * @param requestLocal                  container for the stateful instances scoped to this request -- this must correspond to the
+   *                                      thread calling this method
    * @param actionQueue                   the action queue to use. ReplicaManager#defaultActionQueue is used by default.
+   * @param verificationGuards            the mapping from topic partition to verification guards if transaction verification is used
+   * @param preAppendErrors               the mapping from topic partition to LogAppendResult for errors that occurred before appending
```

Comment:
This seems a little strange. If the partitions have already failed, then why do we need to pass them through `appendRecords`? I would expect instead that the caller would just join the pre-append failures with the append failures.

Reply:
I was considering a write where some partitions were verified and some were not. We would allow the ones that succeeded to still write and not the ones that failed verification.

Not sure if we support multi-partition writes in the produce api, but it appears we do.

Reply:
Oh I see what you are saying -- basically the appendCallback should be defined so the verification errors are joined whenever the callback is defined (ideally after we get the verification errors).

So the callback for verification would include the definition of the append callback. I think this could work.

Time to change all the mock code 😂

Reply:
This is updated.